### PR TITLE
[Feat] Add Support for BertForMaskedLM to vLLM

### PR DESCRIPTION
**Pull request by atalhens**
### Add Support for `BertForMaskedLM` to vLLM

This PR adds support for `BertForMaskedLM` by introducing a dedicated model wrapper that integrates with vLLM's encoder-only execution path and exposes per-token vocabulary logits through the masked language modeling (MLM) head.

## Purpose

vLLM currently supports BERT encoder models but does not support `BertForMaskedLM`, preventing Hugging Face masked language modeling checkpoints from being loaded and executed. This PR closes that gap so users can run MLM checkpoints (e.g. `google-bert/bert-base-uncased`) directly on vLLM.

### What's included

- A `BertForMaskedLM` wrapper (`encoder_only` attention, token-level pooling) that runs the `BertModel` backbone and projects the encoder output through the MLM head to produce per-token logits over the vocabulary, exposed via the
  `token_classify` pooling task.
- Weight loading that handles the full range of real BERT checkpoints:
  - Legacy `gamma`/`beta` LayerNorm parameter names are renamed to
    `weight`/`bias` (original google-bert lineage).
  - The next-sentence-prediction head (`cls.seq_relationship.*`) is dropped, as
    it is not part of the MLM architecture (mirrors HF's own load behavior).
  - The MLM decoder is tied to the input embeddings when the checkpoint relies
    on `tie_word_embeddings` and ships no explicit decoder weight.
- Registry entries for the new architecture.

## Test Plan

- **Unit / correctness test:** `tests/models/language/pooling/test_mlm.py` compares vLLM's `token_classify` output against Hugging Face's `AutoModelForMaskedLM` on `google-bert/bert-base-uncased`, asserting per-token top-1 agreement and closeness of the full vocabulary distributions.

  ```bash
  pytest tests/models/language/pooling/test_mlm.py
  ```
- End-to-end (/pooling): serve the model and query the pooling endpoint, or run offline via LLM.encode(..., pooling_task="token_classify"), then take the argmax at the masked position.
---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [x] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: tests/models/language/pooling/test_mlm.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (DarkLight1337) [inline tests/models/language/pooling/test_mlm.py]: Would it make more sense to just add this model to `test_token_classification.py`?

Comment (atalhens) [reply tests/models/language/pooling/test_mlm.py]: Yeah, I think we can do that. Moved it to `test_token_classification.py`

**Review by DarkLight1337 — APPROVED** [review]
